tests/unit/MigrationTest.js

```diff
@@ -639,4 +640,72 @@ describe('Migrations', () => {
                     });
                 }));
     });
+
+    describe('CheckPreviousReportActionID', () => {
```

Comment:
The tests are very limited to only solve the happy case. Can we add some corner case tests like
{reportAction1: null} or {reportAction1: {}} ?

Reply:
Okay, adding. What would be the scenario in which a `reportAction` is null or an empty object? Since an 'action' object is typically expected to have content, I am guessing that `reportActions` are not in the right condition. Should the check for `previousReportActionID` then fail (and proceed with removing all report actions)? As the migration did not find the field.

Reply:
@roryabraham What should we do in case of zoombie reportActions? like {reportAction1: null} or {reportAction1: {}}.
Should we loop through to find other valid reportActions and check previousReportActionID or just clear all?

Reply:
I agree we should have some edge case handling for zombie data and the like, just so that this code is nice and robust.

> What should we do in case of zoombie reportActions? like {reportAction1: null} or {reportAction1: {}}.
Should we loop through to find other valid reportActions and check previousReportActionID or just clear all?

I would suggest that we loop through until a reportAction w/ a `reportActionID` field is found, then proceed with the existing logic.

Reply:
@neonbhai Could you kindly change this and let me know? So that I can review again?

Reply:
all done!